test/fixtures.rb

```diff
@@ -1300,3 +1300,110 @@ def call(params)
     raise TestException.new
   end
 end
+
+module CookiesAndSessions
```

Comment:
Nitpicking: i'd call this module and corresponding test: ```SessionWithCookies```
Like that it is in sync with ```SessionWithoutCookies```